packages/expect/src/jestMatchersObject.ts

```diff
@@ -59,19 +54,19 @@ export const setMatchers = (
       // expect is defined
 
       class CustomMatcher extends AsymmetricMatcher {
-        sample: Array<any>;
+        sample: Array<unknown>;
 
-        constructor(inverse: boolean = false, ...sample: Array<any>) {
+        constructor(inverse: boolean = false, ...sample: Array<unknown>) {
           super();
           this.inverse = inverse;
           this.sample = sample;
         }
 
-        asymmetricMatch(other: any) {
-          const {pass} = ((matcher(
-            (other: any),
-            ...(this.sample: any),
-          ): any): SyncExpectationResult);
+        asymmetricMatch(other: unknown) {
+          const {pass} = matcher(
+            other,
+            ...(this.sample as [unknown, unknown]),
+          ) as SyncExpectationResult;
```

Comment:
Just copying over from the original code. This typing seems more accurate than my previous change. comments are appreciated